libs/src/sdk/services/EntityManager/EntityManager.ts

```diff
@@ -107,8 +101,63 @@ export class EntityManager implements EntityManagerService {
 
     const jsonResponse = await response.json()
 
+    if (!skipConfirmation) {
+      await this.confirmWrite({
+        blockHash: jsonResponse.receipt.blockHash,
+        blockNumber: jsonResponse.receipt.blockNumber,
+        confirmationTimeout
+      })
+    }
+
     return {
       txReceipt: jsonResponse.receipt
```

Comment:
nit: have this return `blockHash` and `blockNumber` and expose `confirmWrite` so a user could confirm later?
Use case could be something like doing different actions on write success vs confirmed. It doesn't use entityManager but in tipping we just had a use case where we let the user know their tip was sent successfully before it's indexed, but then have some side-effects like unlocking premium content and checking DM permissions when it finally gets indexed on DN

Reply:
Good idea